periphery action: add flag to control `--strict` mode

lukaskollmer opened the pull request:
# periphery action: add flag to control `--strict` mode

## :recycle: Current situation & Problem
The periphery action currently always runs in `--strict` mode, which turns all warnings into errors (or, rather, if any warnings are encountered, the command will fail).
This behaviour is desired in the vast majority of cases, but there are exceptions.
(eg, if you're implementing a soon-to-be-used-but-currently-still-unused component, it would be valid to temporarily have technically unused code in the codebase.)
This flag allows for supporting this.


## :gear: Release Notes 
n/a

## :books: Documentation
n/a

## :white_check_mark: Testing
n/a

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

lukaskollmer commented:
@PSchmiedmayer i'm not really sure whether this is the correct approach (or whether this is even something we'd want to allow); please give some feedback wrt whether this is how one would implement such a flag in the workflow.yml file

(also, feel free to close the PR if we don't want to allow disabling the check this easily)

PSchmiedmayer reviewed (approved):
Sounds good to me; can you ensure that you test this from a different repo by pointing the GitHub Action to this branch to test that it works (without the flag and with setting the flag) before merging this?

Generally fine with this; we should not enable it by default and only make it an exception. But better than removing it completely.

PSchmiedmayer commented:
Also seems like one of the link is throwing an 404; might be good to look into this.

lukaskollmer commented:
@PSchmiedmayer would you say that `cond && a || b` is the correct approach for doing a ternary expression in the script command? or is there some better approach i wasn't able to find?

lukaskollmer commented:
@PSchmiedmayer i ran some tests on this (specify flag as true, specify it as false, omit it entirely) from the MHC repo and it all worked as expected. i also fixed the link and added the PR template rewording changes from the Spezi org.
